auto-instrumentation/okhttp/okhttp-3.0/library/src/main/java/io/opentelemetry/instrumentation/library/okhttp/v3_0/OkHttpInstrumentationConfig.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.library.okhttp.v3_0;
+
+import io.opentelemetry.instrumentation.api.internal.HttpConstants;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nonnull;
+
+/** Configuration for automatic instrumentation of okhttp requests. */
+public final class OkHttpInstrumentationConfig {
+    private static List<String> capturedRequestHeaders = new ArrayList<>();
+    private static List<String> capturedResponseHeaders = new ArrayList<>();
+    private static Set<String> knownMethods = new HashSet<>();
+    private static Map<String, String> peerServiceMapping = new HashMap<>();
+    private static boolean emitExperimentalHttpClientMetrics;
+
+    private OkHttpInstrumentationConfig() {}
+
+    /**
+     * Configures the HTTP request headers that will be captured as span attributes as described in
+     * <a
+     * href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-request-and-response-headers">HTTP
+     * semantic conventions</a>.
+     *
+     * <p>The HTTP request header values will be captured under the {@code
+     * http.request.header.<name>} attribute key. The {@code <name>} part in the attribute key is
+     * the normalized header name: lowercase, with dashes replaced by underscores.
+     *
+     * @param requestHeaders A list of HTTP header names.
+     */
+    public static void setCapturedRequestHeaders(@Nonnull List<String> requestHeaders) {
```

Comment:
Got it, I wasn't aware of it, I'll make the changes

Reply:
Just to confirm something though, after checking it a bit further I remembered that these methods are supposed to be called by our users, I was thinking that if they're not using nullaway in their projects, then this annotation could let their IDEs know that they shouldn't pass nulls here... Do you think we should still remove them for public functions? We might also want to validate null params now that I think about it in case our users decide to pass null regardless of any warnings, though I'm not sure if that'd be too much.

Reply:
I'd recommend applying `@ParametersAreNonnullByDefault` similar to https://github.com/open-telemetry/opentelemetry-java/blob/9a931556fb05dadc1419736d3fecaec96af5f18c/api/all/src/main/java/io/opentelemetry/api/package-info.java#L12-L15

Reply:
Cool! I've applied the changes, thanks